miner/miner.go

```diff
@@ -334,14 +346,45 @@ func (m *Miner) computeTicket(ctx context.Context, addr address.Address, base *M
 	}, nil
 }
 
-func (m *Miner) createBlock(base *MiningBase, addr address.Address, ticket *types.Ticket, proof *types.EPostProof) (*types.BlockMsg, error) {
+func (m *Miner) actorLookup(ctx context.Context, addr address.Address, ts *types.TipSet) (uint64, *types.BigInt, error) {
+	// TODO: strong opportunities for some caching in this method
+	act, err := m.api.StateGetActor(ctx, addr, ts)
+	if err != nil {
+		return 0, nil, xerrors.Errorf("looking up actor failed: %w", err)
+	}
 
-	pending, err := m.api.MpoolPending(context.TODO(), base.ts)
+	msgs, err := m.api.ChainGetTipSetMessages(ctx, ts.Key())
 	if err != nil {
-		return nil, xerrors.Errorf("failed to get pending messages: %w", err)
+		return 0, nil, xerrors.Errorf("failed to get tipset messages: %w", err)
+	}
+
+	balance := act.Balance
+	curnonce := act.Nonce
+
+	sort.Slice(msgs, func(i, j int) bool { // TODO: is this actually needed?
+		return msgs[i].Message.Nonce < msgs[j].Message.Nonce
+	})
+
+	max := int64(-2)
+
+	for _, m := range msgs {
+		if m.Message.From == addr {
+			max = int64(m.Message.Nonce)
+			balance = types.BigSub(balance, m.Message.RequiredFunds())
+		}
 	}
 
-	msgs, err := selectMessages(context.TODO(), m.api.StateGetActor, base, pending)
+	max++ // next unapplied nonce
+
+	if max != -1 && uint64(max) != curnonce {
+		return 0, nil, xerrors.Errorf("tipset messages from %s have too low nonce %d, expected %d, h: %d", addr, max, curnonce, ts.Height())
```

Comment:
Do we want to keep all this as a sanity check?

Reply:
lets drop it